code/modules/spells/general/ring_of_fire.dm

```diff
@@ -0,0 +1,97 @@
+/spell/aoe_turf/ring_of_fire
+	name = "Ring of Fire"
+	desc = "Summon a stationary ring of flames around your current location for 10 seconds. While the ring is active, you are fully immune to fire and burns."
+	school = "conjuration"
+	charge_max = 300
+	cooldown_min = 100
+
+	spell_levels = list(Sp_SPEED = 0, Sp_MOVE = 0)
+	level_max = list(Sp_TOTAL = 4, Sp_SPEED = 3, Sp_MOVE = 1)
+
+	spell_flags = NEEDSCLOTHES
+	charge_type = Sp_RECHARGE
+	invocation = "E ROHA"
+	invocation_type = SpI_SHOUT
+	hud_state = "wiz_firering"
+
+	duration = 100
+	range = 3
+	var/move_with_user = 0
+
+/spell/aoe_turf/ring_of_fire/choose_targets(mob/user = usr)
+	return trange(range, get_turf(user)) - trange(range - 1, get_turf(user))
+
+/spell/aoe_turf/ring_of_fire/cast(list/targets, mob/user)
+
+	var/obj/effect/ring = new /obj/effect/ring_of_fire(get_turf(user), targets, duration)
+
+	if(move_with_user)
+		user.lock_atom(ring, /datum/locking_category/ring_of_fire)
+
+	to_chat(user, "<span class='danger'>You summon a ring of fire around yourself.</span>")
+
+	if(isliving(user))
+		var/mob/living/L = user
+
+		if(!L.mutations.Find(M_RESIST_HEAT))
+			to_chat(L, "<span class='info'>You feel resistant to fire.</span>")
+
+		L.mutations.Add(M_RESIST_HEAT)
```

Comment:
If you already have heat resistance, your mutations list will have a duplicate M_RESIST_HEAT in it. That duplicate is then removed after 10 seconds.

If that was tabbed, you'd lose heat resistance from eating ryetalin (if you had the heat resistant gene beforehand, which gives you the same mutation)